google-cloud-pubsub/lib/google/cloud/pubsub/topic/batch_publisher.rb

```diff
@@ -57,9 +57,11 @@ def publish data, attributes = {}
 
           ##
           # @private Create Message objects with message ids.
-          def to_gcloud_messages message_ids
+          def to_gcloud_messages message_ids, published_at = Time.now
+            publish_time = Convert.time_to_timestamp published_at
             msgs = @messages.zip(Array(message_ids)).map do |msg, id|
               msg.message_id = id
+              msg.publish_time = publish_time
```

Comment:
Might be better to use `published_at` instead of the alias `publish_time`, to make it less likely that some maintainer misses this when searching for usages. Or reverse which is the impl and which is the alias.